pkg/scheduler/factory.go

```diff
@@ -171,9 +180,13 @@ func (c *Configurator) createFromProvider(providerName string) (*Scheduler, erro
 		return nil, fmt.Errorf("algorithm provider %q is not registered", providerName)
 	}
 
-	// Combine the provided plugins with the ones from component config.
-	defaultPlugins.Apply(c.plugins)
-	c.plugins = defaultPlugins
+	for i := range c.profiles {
+		prof := &c.profiles[i]
+		plugins := &schedulerapi.Plugins{}
+		plugins.Append(defaultPlugins)
+		plugins.Apply(prof.Plugins)
+		prof.Plugins = plugins
```

Comment:
nit: these 5 lines can be extracted into a function to avoid duplication (with L304~L308)

Reply:
I wouldn't mind too much. They are slightly different and `createFromConfig` should go away soon when we remove policy support.